### PR TITLE
GRAL-2489: Add sonarqube related files and scripts

### PR DESCRIPTION
**Pull request by RuTsEST**

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Required metadata
```
Comment (estevanjantsk) [inline sonar-project.properties:1]: As you can see from this page at Step 2 https://pipedrive.atlassian.net/wiki/spaces/SD/pages/12616076/How+to+add+a+project+to+SonarQube

There are some missing props, for ex: 
```
# Path to the source code directory relative to the sonar-project.properties file.
sonar.sources=. 
```
Is this ok?

**Review by estevanjantsk — COMMENTED** [review]
Just one comment :). Other than that lgtm.

File: sonar-project.properties (hunk 1 above)
Comment (RuTsEST) [reply sonar-project.properties:1]: I thought that the required metadata are only:
```
sonar.projectKey=Pipedrive:Green:dogfood
sonar.projectName=Dogfood
```
But yeah, I do agree that the sources should probably be specified

Comment (RuTsEST) [reply sonar-project.properties:1]: Just ran with the sources defined and the output was pretty much the same, I guess I'll specify it since it seems to be best practice but I don't think it is that relevant TBH.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,9 @@
     "test": "npm run test:functional:start && npm run test:functional && npm run test:functional:stop",
     "test:functional": "node ./test/functional/environment.js",
     "test:functional:start": "node ./test/functional/environment.js --start-environment",
-    "test:functional:stop": "node ./test/functional/environment.js --stop-environment"
+    "test:functional:stop": "node ./test/functional/environment.js --stop-environment",
+    "sonar": "npm run test:coverage && pipedrive-sonar-scanner",
```
Comment (AndreasSaltsberg) [inline package.json:14]: Could this be separate? so sonar command only runs the scanner?

File: client-nodejs.iml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="WEB_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (AndreasSaltsberg) [inline client-nodejs.iml:9]: Why is this file needed? Should it be in git?

Comment (RuTsEST) [reply client-nodejs.iml:9]: I think it's not needed, I'll move to gitignore

File: package.json (hunk 2 above)
Comment (RuTsEST) [reply package.json:14]: But wouldn't it need the coverage generated by `test:coverage`?

Comment (AndreasSaltsberg) [reply package.json:14]: where is `sonar` command executed? If it does it in jenkins pipeline, won't it run tests twice in that case?

Comment (RuTsEST) [reply package.json:14]: I'm not sure where it's executed, I think pipeline yes. I'll separate these.

**Review by AndreasSaltsberg — APPROVED** [review]
LGTM
